unit-tests/non_func/test-leak-detection.cpp

```diff
@@ -0,0 +1,395 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2015 Intel Corporation. All Rights Reserved.
+
+//#cmake: static!
+
+
+#define CATCH_CONFIG_MAIN
+#include "../catch.h"
+#include "../unit-tests-common.h"
+using namespace librealsense;
+
+
+
+//#include "../approx.h"
+#include <cmath>
+//#include <iostream>
+//#include <chrono>
+//#include <ctime>
+//#include <algorithm>
+//#include <librealsense2/rs.hpp>
+#include <librealsense2/hpp/rs_sensor.hpp>
+//#include "../../common/tiny-profiler.h"
+#include "./../src/environment.h"
+
+#include "../unit-tests-common.h"
+#include <numeric>
+#include <stdlib.h> 
+
+using namespace librealsense;
+using namespace librealsense::platform;
+
+constexpr int ITERATIONS_PER_CONFIG = 100;
+constexpr int INNER_ITERATIONS_PER_CONFIG = 10;
+constexpr int DELAY_INCREMENT_THRESHOLD = 3; //[%]
+constexpr int DELAY_INCREMENT_THRESHOLD_IMU = 8; //[%]
+constexpr int SPIKE_THRESHOLD = 2; //[stdev]
+
+// Input:     vector that represent samples of delay to first frame of one stream
+// Output:  - vector of line fitting data according to least squares algorithm 
+//          - slope of the fitted line
+// reference: https://en.wikipedia.org/wiki/Least_squares
+double line_fitting(const std::vector<double>& y_vec, std::vector<double>& y_fit = std::vector<double>())
```

Comment:
The output has a default?
What's the return value? It's not documented...

Reply:
return value is the output as written (updated it to return slope)
default value is (added)